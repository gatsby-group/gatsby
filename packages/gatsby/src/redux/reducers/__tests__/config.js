const reducer = require(`../config`)

describe(`config reducer`, () => {
  it(`let's you add a config`, () => {
    const action = {
      type: `SET_SITE_CONFIG`,
      payload: {
        siteMetadata: {
          hi: true,
        },
      },
    }
    expect(reducer(undefined, action)).toMatchSnapshot()
  })

  it(`handles empty configs`, () => {
    const action = {
      type: `SET_SITE_CONFIG`,
    }
    expect(reducer(undefined, action)).toMatchSnapshot()
  })

  it(`It corrects pathPrefixes without a forward slash at beginning`, () => {
    const action = {
      type: `SET_SITE_CONFIG`,
      payload: {
        pathPrefix: `prefix`,
      },
    }
    expect(reducer(undefined, action).pathPrefix).toBe(`/prefix`)
  })

  it(`It removes trailing forward slash`, () => {
    const action = {
      type: `SET_SITE_CONFIG`,
      payload: {
        pathPrefix: `/prefix/`,
      },
    }
    expect(reducer(undefined, action).pathPrefix).toBe(`/prefix`)
  })
})
