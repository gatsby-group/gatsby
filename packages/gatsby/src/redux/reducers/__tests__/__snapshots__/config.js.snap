// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config reducer Validates configs with unsupported options 1`] = `"The site's gatsby-config.js failed validation"`;

exports[`config reducer handles empty configs 1`] = `
Object {
  "pathPrefix": "",
  "polyfill": true,
}
`;

exports[`config reducer let's you add a config 1`] = `
Object {
  "pathPrefix": "",
  "polyfill": true,
  "siteMetadata": Object {
    "hi": true,
  },
}
`;

exports[`config reducer throws with a suggestion when an invalid key is passed 1`] = `"[1m[31mplugin is not allowed. Did you mean \\"plugins\\"?[39m[22m"`;
